### PR TITLE
Able to use server-native image (master)

### LINKED ISSUE #473: Able to use server-native image
Operator is not able to run native image due following JDK options are not supported on it
- MaxRAM
- UseSerialGC
- MinHeapFreeRatio
- MaxHeapFreeRatio

Image: quay.io/infinispan/server-native:11.0.1.Final

We should detect/define that operator is running native image and don't set these options by default
@rigazilla wdyt?

Thanks the colleague of mine, @dsevost which is finding this during Quarkus demo preparation
https://github.com/e2e-openshift-demos/quarkus-demo/blob/b-step-03_4/jdg/datagrid-operator-setup.sh#L58

### PR DESCRIPTION
**Pull request by dmvolod**
Fixes #473

### CODE REVIEW

File: pkg/apis/infinispan/v1/types_util.go [hunk 1]
```diff
@@ -13,6 +14,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+type ImageType string
+
+const (
+	// Container image based on JDK
+	ImageTypeJava ImageType = "Java"
```
Comment (ryanemerson) [inline pkg/apis/infinispan/v1/types_util.go:21]: Nitpick, but should we call this "JVM"? Both the normal and native images are written in Java, this just determines the runtime.

**Review by ryanemerson — COMMENTED** [review]
Overall LGTM, just one small thing.

File: pkg/apis/infinispan/v1/types_util.go (hunk 1 above)
Comment (dmvolod) [reply pkg/apis/infinispan/v1/types_util.go:21]: Yes, good point, @ryanemerson . Will fix.

**Review by ryanemerson — APPROVED** [review]

**Review by rigazilla — APPROVED** [review]

**Comment by dmvolod** [comment]
Thanks for review, @rigazilla and @ryanemerson 
Merged

**Pull request merged**